app/controllers/attributes_controller.rb

```diff
@@ -2,35 +2,23 @@ class AttributesController < ApplicationController
   include AuthenticatedApiConcern
 
   def show
-    local_attributes, remote_attributes = get_attributes_from_params(
-      params.fetch(:attributes),
-      permission_level: :get,
-    )
+    attributes = params.fetch(:attributes)
+    validate_attributes!(attributes, :get)
 
-    values = @govuk_account_session.get_local_attributes(local_attributes)
-      .merge(@govuk_account_session.get_remote_attributes(remote_attributes))
-
-    render_api_response values: values.compact
+    render_api_response values: @govuk_account_session.get_attributes(attributes)
   end
 
   def update
-    local_attributes, remote_attributes = get_attributes_from_params(
-      params.fetch(:attributes).permit!.to_h,
-      permission_level: :set,
-      is_hash: true,
-    )
-
-    @govuk_account_session.set_local_attributes(local_attributes)
-    @govuk_account_session.set_remote_attributes(remote_attributes)
+    attributes = params.fetch(:attributes).permit!.to_h
```

Comment:
Humm, does `permit!` here undemine the strong attributes?
I wonder if its worth rejecting anything that isn't in a known structure?

Reply:
Ok fair enough, That's a well point well made!